control/freqplot.py

```diff
@@ -565,55 +572,61 @@ def nyquist_plot(syslist, omega=None, plot=True, label_freq=0,
         label_freq = kwargs.pop('labelFreq')
 
     # If argument was a singleton, turn it into a list
-    if not getattr(syslist, '__iter__', False):
+    if not hasattr(syslist, '__iter__'):
         syslist = (syslist,)
 
     # Select a default range if none is provided
     if omega is None:
-        omega = default_frequency_range(syslist)
-
-    # Interpolate between wmin and wmax if a tuple or list are provided
-    elif isinstance(omega, list) or isinstance(omega, tuple):
-        # Only accept tuple or list of length 2
-        if len(omega) != 2:
-            raise ValueError("Supported frequency arguments are (wmin,wmax)"
-                             "tuple or list, or frequency vector. ")
-        omega = np.logspace(np.log10(omega[0]), np.log10(omega[1]),
-                            num=50, endpoint=True, base=10.0)
-
-    for sys in syslist:
-        if sys.ninputs > 1 or sys.noutputs > 1:
-            # TODO: Add MIMO nyquist plots.
-            raise NotImplementedError(
-                "Nyquist is currently only implemented for SISO systems.")
+        if omega_limits is None:
+            # Select a default range if none is provided
+            omega = default_frequency_range(syslist, Hz=False,
```

Comment:
Should this be an internal function `_default_frequency_range`?

Reply:
probably. its counter part, `_default_time_vector` is. I'll change it.